Yoast/wpseo-woocommerce
# Send data to worker on various variation manipulations

## Context


* Lingo requires data to be scraped, prepared, and sent to the worker.

## Summary

This PR can be summarized in the following changelog entry:

* Scrapes data about global and variation identifiers from the DOM and sends it to the worker when needed.

## Relevant technical choices:

**IMPORTANT**
* Since we cannot hook into certain bulk events (since they use a window.prompt or window.confirmation) we cannot be certain about what the user accomplished. In these cases we invalidate the data we send to the worker by setting `variantIdentifierDataIsValid` to false. The lingo team will interpret this as a "Grey" score and ask the user to reload.
* Upon reloading all data is fetched via the woocommerce php side methods, which can be trusted again :) 

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

Setup:
* Have Free checked out in trunk and change the `isApplicable` function return `true` in its `packages\yoastseo\src\scoring\assessments\seo\ProductIdentifiersAssessment.js` file and built
* Go to Woo's vendor in the Free directory and change the `isApplicable` function return `true` in its `vendor\yoast\wordpress-seo\packages\yoastseo\src\scoring\assessments\seo\ProductIdentifiersAssessment.js` file
* Build the vendor Free plugin
* Also build the WooCommerce plugin

(Test instructions borrowed by https://yoast.atlassian.net/browse/PC-144 and enhanced):
* Add a new product
* Confirm that the Product identifiers assessment shows up in the SEO assessments with an orange bullet and the following feedback: `Product identifier: Your product is missing a product identifier (like a GTIN code). Include this if you can, as it will help search engines to better understand your content.`
* Hover over the links in the feedback strings. Confirm that the first link starts with `https://yoa.st/4ly` and the second one with `https://yoa.st/4lz`
* Go to the Yoast SEO tab in the product metabox and add a product identifier.
* Confirm that the Product identifiers assessment returns a green bullet with the following feedback: `Product identifier: Good job!`
* Remove the product indentifier from the metabox and confirm that the assesment returns back in its orange state. Add the product identifier back.
* Make the product a variable product and add variants to the product:  first click on the Attributes tab and add some attributes (make sure to check the `Used for variations`. Then click on the Variants tab that appears and create variants from the attributes. Make sure you have created enough variants (over 15) so that we can have multiple paginated pages in the variants tab.
* When the variants are created, we should still get the green bullet in the assessment because we haven't yet put a price in any variant.
* Add a price to at least one variant and confirm that the Product identifiers assessment returns an orange bullet with the following feedback: ` Product identifier: Not all your product variants have a product identifier. Include this if you can, as it will help search engines to better understand your content.`. Also confirm that the links direct to the URLs mentioned above.
* Add a product identifier to each variant that has a price: click to expand each variants, and scroll down to the Yoast SEO options.
* As soon as all variants that have a price have at least one identifier confirm that the Product identifiers assessment returns a green bullet with the following feedback: `Product identifier: Good job!`
* Delete one variant with a price and indentifiers and confirm that the assessment stays into the green state
* Remove all identifiers from at least one variant with a price and confirm that the assessment went back to its orange state.
* Delete that variant and confirm that the assessment goes back to green state.
* -- To be tested when [the accompaniying Free PR](https://yoast.atlassian.net/browse/PC-241) is implemented: When clicking one of the `Delete all variations`, `Set regular prices`, `Increase regular prices`, `Decrease regular prices` option of the Variations tab's dropdown and clicking `Go`, the assesment should turn grey. That grey assesment should be turned to the appropriate color once reloading.
* Change the product to a cornerstone product (enable the Cornerstone content toggle).
* Confirm that the Product identifiers assessment still appears and that the feedback hasn’t changed.
* Change your site language to a language other than English
* Confirm that the Product identifiers assessment still appears (the string will still be in English until the strings for that language show up as translated on https://translationspress.com/ )
* Activate the Premium plugin.
* Go back to the product and make sure that the Product identifiers assessment still appears. 
* Trigger another condition of the assessment and make sure that you get the expected result.
* Enable the Classic editor plugin.
* Go back to the product and make sure that the Product identifiers assessment still appears. 


### Test instructions for QA when the code is in the RC


* [] QA should use the same steps as above.


QA can test this PR by following these steps:

* QA should follow the Test Instructions of [the accompanying Free PR](https://yoast.atlassian.net/browse/PC-241)

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes PC-138
